CARL/source/Recognizer.cpp

```diff
@@ -28,24 +28,20 @@ namespace carl::action
         // the only place with a reasonable idea of expected variance (this is sort of a way to allow
         // descriptors to self-normalize their distance space). Second, an alternate creation path should be
         // introduced for static pose gestures where the system dices up a longer segment into chunks.
-        constexpr double kAverageDistanceEpsilon{ 0.000000001 };
+        constexpr NumberT kAverageDistanceEpsilon{ 0.000000001 };
 
         // Slightly less naive holistic resampling. Still linear, though.
-        action::Recording resample(const action::Example& original, double frameDuration)
+        std::vector<InputSample> resample(gsl::span<const InputSample> samples, NumberT startTimestamp, NumberT endTimestamp, NumberT frameDuration)
         {
-            auto samples = original.getRecording().getSamples();
             size_t samplesIdx = 0;
 
             std::vector<InputSample> newSamples{};
-            size_t newSamplesCount = std::max<size_t>(
-                static_cast<size_t>(
-                    (original.getEndTimestamp() - original.getStartTimestamp()) / frameDuration),
-                1);
+            size_t newSamplesCount = std::max<size_t>(static_cast<size_t>(std::ceil((endTimestamp - startTimestamp) / frameDuration)), 1);
```

Comment:
It's subtle, but the inclusion of `std::ceil` in this equation is also a significant bug fix. Without that, the last sample of a resampling will typically be from *before* `endTimestamp`, which in very short actions at slow framerates can cause the final pose of the action to be "sampled out." By adding `ceil`, an additionally sample is typically introduced, ensuring that the final pose of the example is represented in the resampling.